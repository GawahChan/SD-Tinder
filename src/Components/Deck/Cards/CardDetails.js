import React, { Component } from 'react';
import { CardTitle, CardSubtitle, CardInfoButton } from './style';

class CardDetails extends Component {
    constructor(props) {
        super(props);
        this.state = { displayBio: false }
    }

    toggleBio = () => this.setState({ displayBio: !this.state.displayBio })

    render() {
        const { stuntDouble } = this.props;
        return (
            <div>
                <CardTitle>
                    {stuntDouble.name}, {stuntDouble.age}
                </CardTitle>
                <CardSubtitle>
                    {stuntDouble.location}
                    <CardInfoButton onTap={this.toggleBio} />
                </CardSubtitle>
                {this.state.displayBio && <p>{stuntDouble.bio}</p>}
            </div >
        );
    }
}

export default CardDetails;