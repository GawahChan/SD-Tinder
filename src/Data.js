const Data = [
    {id: 0, name: 'Mila Kunis', age: '30', location: 'LA', bio: 'Actress' }
];

export default Data;