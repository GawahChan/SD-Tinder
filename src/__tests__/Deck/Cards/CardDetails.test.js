import React from 'react';
import { shallow } from 'enzyme';
import CardDetails from '../../../Components/Deck/Cards/CardDetails';
import Data from '../../../Data';

describe('CardDetails component', () => {
    let data = [Data[0]];
    let props = { stuntDouble: data };
    let cardDetails = shallow(<CardDetails {...props} />);

    it('renders correctly', () => {
        expect(cardDetails).toMatchSnapshot();
    });

    it('initialises `displayBio` state to false', () => {
        expect(cardDetails.state('displayBio')).toBe(false);
    });

    describe('toggleBio event handler', () => {

        beforeEach(() => {
            cardDetails.instance().toggleBio();
        })

        afterEach(() => {
            cardDetails.setState({ displayBio: false });
        });

        it('user clicks info button once, sets `displayBio` state to "true"', () => {

            expect(cardDetails.state('displayBio')).toBe(true);
        });

        it('user clicks info button twice, sets `displayBio` state to "false"', () => {
            cardDetails.instance().toggleBio();

            expect(cardDetails.state('displayBio')).toBe(false);
        });
    });
});