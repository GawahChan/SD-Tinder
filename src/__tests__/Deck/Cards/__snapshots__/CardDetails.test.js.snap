// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails component renders correctly 1`] = `
<div>
  <styled.h1>
    , 
  </styled.h1>
  <styled.div>
    <Styled(Component)
      onTap={[Function]}
    />
  </styled.div>
</div>
`;
